src/handlers/transfers/handler.js

```diff
@@ -91,53 +98,109 @@ const prepare = async (error, messages) => {
     if (validationPassed) {
       Logger.info('TransferService::prepare::validationPassed')
-      const existingTransfer = await TransferService.getTransferById(payload.transferId)
-      if (!existingTransfer) {
+      try {
         Logger.info('TransferService::prepare::validationPassed::newEntry')
+        // Save the valid transfer into the database
         await TransferService.prepare(payload)
-        if (!Kafka.Consumer.isConsumerAutoCommitEnabled(kafkaTopic)) {
-          await consumer.commitMessageSync(message)
-        }
-        // position topic to be created and inserted here
-        await Utility.produceParticipantMessage(payload.payerFsp, TransferEventType.POSITION, TransferEventAction.PREPARE, message.value, Utility.ENUMS.STATE.SUCCESS)
-        return true
-      } else {
-        Logger.info('TransferService::prepare::validationFailed::existingEntry')
+      } catch (err) {
+        Logger.info('TransferService::prepare::validationPassed::duplicate found while inserting into transfer table')
         if (!Kafka.Consumer.isConsumerAutoCommitEnabled(kafkaTopic)) {
           await consumer.commitMessageSync(message)
         }
         // notification of duplicate to go here
-        message.value.content.payload = Utility.createPrepareErrorStatus(errorCode, errorDescription, message.value.content.payload.extensionList)
-        await Utility.produceGeneralMessage(Utility.ENUMS.NOTIFICATION, Utility.ENUMS.EVENT, message.value, Utility.createState(Utility.ENUMS.STATE.FAILURE.status, errorCode, errorDescription))
+        Logger.info('TransferService::prepare::validationPassed::send the callback notification for duplicate request')
+        // send generic internal error
+        message.value.content.payload = Utility.createPrepareErrorStatus(errorCodeInternal, errorDescriptionInternal, message.value.content.payload.extensionList)
+        await Utility.produceGeneralMessage(Utility.ENUMS.NOTIFICATION, Utility.ENUMS.EVENT, message.value, Utility.createState(Utility.ENUMS.STATE.FAILURE.status, errorCodeInternal, errorDescriptionInternal))
         return true
       }
+      if (!Kafka.Consumer.isConsumerAutoCommitEnabled(kafkaTopic)) {
+        await consumer.commitMessageSync(message)
+      }
+      Logger.info('TransferService::prepare::validationPassed::create position topic')
+      // position topic to be created and inserted here
+      await Utility.produceParticipantMessage(payload.payerFsp, TransferEventType.POSITION, TransferEventAction.PREPARE, message.value, Utility.ENUMS.STATE.SUCCESS)
+      return true
     } else {
       Logger.info('TransferService::prepare::validationFailed')
-      // need to determine what happens with existing transfer with a validation failure
-      const existingTransfer = await TransferService.getById(payload.transferId)
-      if (!existingTransfer) {
+      try {
         Logger.info('TransferService::prepare::validationFailed::newEntry')
```

Comment:
Should there not be a consumer.commitMessageSync here for this validationFailed branch?

I only see a commit in the catch, not outside?